js/gateio.js

```diff
@@ -2035,23 +2070,66 @@ module.exports = class gateio extends Exchange {
             //     }
             // }
         }
-        if (type === 'limit') {
-            if (!price) {
-                throw new ArgumentsRequired ('Argument price is required for ' + this.id + '.createOrder for limit orders');
-            }
-            request['price'] = this.priceToPrecision (symbol, price);
-        } else if (type === 'market' && contract) {
+        if (type === 'market' && market['contract']) {
             request['tif'] = 'ioc';
             request['price'] = 0;
+        } else {
+            request['price'] = price;
         }
-        const method = this.getSupportedMapping (market['type'], {
-            'spot': 'privateSpotPostOrders',
-            // 'margin': 'privateSpotPostOrders',
-            'swap': 'privateFuturesPostSettleOrders',
-            'future': 'privateDeliveryPostSettleOrders',
-        });
-        const response = await this[method] (this.extend (request, params));
-        return this.parseOrder (response, market);
+        return this.extend (request, params);
+    }
+
+    stopLossRequest (symbol, type, side, amount, price = undefined, params = {}) {
```

Comment:
We usually don't add a standalone method for it, instead implementing it inside `createOrder`, if the order type is a stop-order or if the stopPrice is available from `params`.

Reply:
Check Binance, for example: https://github.com/ccxt/ccxt/blob/master/js/binance.js#L2338